src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -127,71 +178,125 @@ public Command speaker() {
 
 	public Command preAmp() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.PRE_AMP),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
-			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
 			funnelStateHandler.setState(FunnelState.STOP),
+			elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+			flywheelStateHandler.setState(FlywheelState.DEFAULT),
+			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
 			elevatorStatesHandler.setState(ElevatorStates.PRE_AMP)
-		).until(this::isNoteInElevatorRoller);
+		);
 	}
 
 	public Command amp() {
 		return new ParallelCommandGroup(
-			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)).until(() -> swerve.isAtHeading(Field.getAngleToAmp())),
+			setCurrentStateValue(RobotState.AMP),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+					swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE.withAimAssist(AimAssist.AMP)),
+					elevatorStatesHandler.setState(ElevatorStates.PRE_AMP)
+				).until(this::isReadyToAmp),
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.AMP)
+				).until(() -> !isNoteInElevatorRoller()),
+				new ParallelCommandGroup(
+					elevatorRollerStateHandler.setState(ElevatorRollerState.STOP),
+					elevatorStatesHandler.setState(ElevatorStates.IDLE)
+				)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.AMP),
-			funnelStateHandler.setState(FunnelState.AMP),
+			intakeStatesHandler.setState(IntakeStates.STOP),
 			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.AMP)
+			funnelStateHandler.setState(FunnelState.STOP)
 		);
 	}
 
 	public Command transferShooterElevator() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.TRANSFER_SHOOTER_ELEVATOR),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				),//.until(isReadyToTransfer)
```

Comment:
When is it ready to transfer? Write the function maybe or remove it and fix if u don't think it's necessary